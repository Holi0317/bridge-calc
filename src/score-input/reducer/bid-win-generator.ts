import {fillObj} from '../../utils'

/**
 * Generate default bid/win. I.E. fill in 0
 */
export function bidWinGenerator(playerID: string[]) {
  return fillObj({}, playerID, 0)
}
