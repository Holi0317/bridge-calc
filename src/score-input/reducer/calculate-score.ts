/**
 * Helper function for reducer.
 * Calculate score of this round from given parameters.
 *
 * @param bid - Number of bid stack given
 * @param win - Number of win gain stack
 * @return Score of the player at that round
 */
export function calculateScore(bid: number, win: number): number {
  if (bid === win) {
    return (bid) ** 2 + 10
  } else {
    return -((win - bid) ** 2)
  }
}
