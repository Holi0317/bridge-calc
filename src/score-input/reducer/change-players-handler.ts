import mapValues from 'lodash-es/mapValues'
import range from 'lodash-es/range'
import {toEndedState, toWaitingBidState} from './converter'
import {toFront} from '../../utils'
import {IEndedState, IWaitingBidState, IWaitingWinState} from './types'
import {IChangePlayersAction} from '../actions/change-players'

export function changePlayersHandler(rawState: IWaitingBidState | IWaitingWinState, {newNames, rounds, maker, time}: IChangePlayersAction): IWaitingBidState | IEndedState {
  // Short circuit. When action.rounds is less than current round
  if (rounds < rawState.currentRound) {
    // tslint:disable-next-line: no-shadowed-variable
    const state = toEndedState(rawState, time)
    state.names = newNames
    state.rounds = rounds
    state.scores = mapValues(state.scores, (score: number[]) => score.slice(0, rounds))
    return state
  }

  const state = toWaitingBidState(rawState)
  state.rounds = rounds

  // Set currentPlayerOrder
  const newPlayerIDList = Object.keys(newNames)
  state.currentPlayerOrder = toFront(newPlayerIDList, newPlayerIDList.indexOf(maker))

  // Change scores
  const oldScores = state.scores
  const freshScores = range(state.currentRound - 1).fill(0) // Score for new players. I am terrible at naming.
  state.scores = mapValues(newNames, (name: string, ID: string): number[] => (
    (ID in oldScores)
      ? oldScores[ID]
      : freshScores
  ))

  // Change bid
  const oldBid = state.bid
  state.bid = mapValues(newNames, (name: string, ID: string): number => (
    (ID in oldBid)
      ? oldBid[ID]
      : 0
  ))

  // Change names field lastly
  state.names = newNames

  return state
}
