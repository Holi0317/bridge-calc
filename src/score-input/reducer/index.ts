export { currentGameReducer } from "./reducer";
export { GameState } from "./types";
export {
  isGameState,
  isEndedState,
  isWaitingBidState,
  isWaitingWinState
} from "./validator";
