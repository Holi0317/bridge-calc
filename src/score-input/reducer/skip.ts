import mapValues from 'lodash-es/mapValues'
import {toFront} from '../../utils'
import {toEndedState, toWaitingBidState} from './converter'
import {bidWinGenerator} from './bid-win-generator'
import {IEndedState, IWaitingBidState, IWaitingWinState} from './types'

/**
 * Skip n round(s) of game.
 */
export function skip(state: IWaitingBidState | IWaitingWinState, n: number, time: Date): IWaitingBidState | IEndedState {
  if (state.currentRound > state.rounds) {
    // Last round. Return regardless of content of n.
    return toEndedState(state, time)
  }
  if (n <= 0) {
    // Last skip action and not last round
    return toWaitingBidState(state)
  }

  const newState = toWaitingBidState(state)
  newState.bid = bidWinGenerator(Object.keys(state.names))
  newState.scores = mapValues(state.scores, score => [...score, 0])
  newState.currentPlayerOrder = toFront(state.currentPlayerOrder, 1)
  newState.currentRound = state.currentRound + 1

  return skip(newState, n - 1, time)
}
