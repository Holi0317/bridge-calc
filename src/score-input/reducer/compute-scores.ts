import { mapValues } from "lodash-es";
import { calculateScore } from "./calculate-score";
import { PlayerMap } from "../../types";

/**
 * Helper function for reducer.
 * Compute new scores object by appending ended round score to the end of each player's score array.
 * @param bids - Player map that maps this round's bid
 * @param wins - Player map that maps this round's win
 * @param scores - Original map that maps player ID to score array. This will NOT be mutated.
 * @returns New score object that maps player id to their score array
 */
export function computeScores(
  bids: PlayerMap<number>,
  wins: PlayerMap<number>,
  scores: PlayerMap<number[]>
) {
  return mapValues(scores, (score, playerID) => {
    const bid = bids[playerID];
    const win = wins[playerID];
    return [...score, calculateScore(bid, win)];
  });
}
