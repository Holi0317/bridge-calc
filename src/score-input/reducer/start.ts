import {GameStage} from '../game-stage'
import {skip} from './skip'
import {bidWinGenerator} from './bid-win-generator'
import {fillObj} from '../../utils'

import {IStartAction} from '../../actions/current-game'
import {IWaitingBidState} from './types'

/**
 * Helper function for reducer.
 * Start a new game base on start action parameters.
 * @param action
 */
export function start(action: IStartAction) {
  const playerIDs = Object.keys(action.playerNames)

  const firstState: IWaitingBidState = {
    stage: GameStage.waitingBid,
    rounds: action.rounds,
    startTime: action.startTime,
    names: action.playerNames,
    scores: fillObj({}, playerIDs, []),
    bid: bidWinGenerator(playerIDs),
    currentPlayerOrder: playerIDs,
    currentRound: 1
  }
  return skip(firstState, action.startingRound - 1, action.startTime)
}
