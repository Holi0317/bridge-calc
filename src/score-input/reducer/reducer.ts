import {GameStage} from '../game-stage'
import {skip} from './skip'
import {start} from './start'
import {winHandler} from './win-handler'
import {changePlayersHandler} from './change-players-handler'
import {toWaitingBidState, toWaitingWinState} from './converter'
import {bidWinGenerator} from './bid-win-generator'
import {GameState} from './types'
import {CurrentGameActions} from '../actions'
import {SKIP} from '../actions/skip'
import {SET_BID} from '../actions/set-bid'
import {SET_WIN} from '../actions/set-win'
import {BID} from '../actions/bid'
import {WIN} from '../actions/win'
import {UNDO} from '../actions/undo'
import {CHANGE_PLAYERS} from '../actions/change-players'
import {START} from '../actions/start'
import {REPLACE_CURRENT_GAME} from '../actions/replace-current-game'

const defaultState: GameState = null

export function currentGameReducer(state: GameState = defaultState, action: CurrentGameActions): GameState {
  if (action.type === START) {
    // START action will always return a new state, ignoring the original state
    return start(action)
  }
  if (action.type === REPLACE_CURRENT_GAME) {
    // REPLACE_CURRENT_GAME ignores all safety check
    return action.payload
  }
  if (state == null) {
    // Prevent any other action when game is not initialized
    return null
  }
  if (state.stage === GameStage.ended) {
    // Ended game is read only. No action other than START can mutate it.
    return state
  }

  switch (action.type) {
  case SKIP:
    return skip(state, action.times || 1, action.time)
  case SET_BID:
    return {
      ...state,
      bid: action.payload
    }
  case SET_WIN:
    return {
      ...toWaitingWinState(state),
      win: action.payload
    }
  case BID:
    return {
      ...state,
      stage: GameStage.waitingWin,
      bid: action.payload || state.bid,
      win: bidWinGenerator(Object.keys(state.names))
    }
  case WIN:
    return winHandler(state, action)
  case UNDO:
    if (state.stage === GameStage.waitingBid) {
      return state
    }
    return toWaitingBidState(state)
  case CHANGE_PLAYERS:
    return changePlayersHandler(state, action)
  default:
    return state
  }
}
