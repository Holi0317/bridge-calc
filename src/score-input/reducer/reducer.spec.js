import {replaceCurrentGameAction} from '../actions/replace-current-game'
import {waitingBidState, waitingWinState, endedState, genMap} from '../../../test-fixtures/current-game-states'
import {currentGameReducer as reducer} from './reducer'
import * as lolex from 'lolex'
import {startAction} from '../actions/start'
import {skipAction} from '../actions/skip'
import {setBidAction} from '../actions/set-bid'
import {setWinAction} from '../actions/set-win'
import {bidAction} from '../actions/bid'
import {winAction} from '../actions/win'
import {undoAction} from '../actions/undo'
import {changePlayersAction} from '../actions/change-players'

jest.mock('cuid', () => {
  let count = 0
  return jest.fn(() => '' + count++)
})

let clock = null
beforeEach(() => {
  clock = lolex.install()
})
afterEach(() => {
  if (clock) {
    clock.uninstall()
  }
})

test('Default state should be null', () => {
  const expected = null
  const actual = reducer(undefined, {type: '#NULL'})
  expect(expected).toBe(actual)
})

test('Production of default basic state after start', () => {
  const action = startAction(13, ['John', 'Mary', 'Henry', 'Joe'], 1)
  const actual = reducer(null, action)
  expect(actual).toMatchSnapshot()
})

test('Start on second round should work', () => {
  const action = startAction(13, ['John', 'Mary', 'Henry', 'Joe'], 2)
  const actual = reducer(null, action)
  expect(actual).toMatchSnapshot()
})

test('Replace current game should replace its state when state is undefined', () => {
  const gameState = {
    ...waitingBidState
  }
  const action = replaceCurrentGameAction(gameState)
  const actual = reducer(null, action)
  const expected = {
    ...waitingBidState
  }
  expect(actual).toEqual(expected)
})

test('Replace current game should work even current state is ended', () => {
  const state = {
    ...endedState
  }
  const gameState = {
    ...waitingWinState
  }
  const action = replaceCurrentGameAction(gameState)
  const actual = reducer(state, action)
  const expected = {
    ...waitingWinState
  }
  expect(actual).toEqual(expected)
})

test('Skip should do no-op when the state is null', () => {
  const expected = null
  const action = skipAction(1)
  const actual = reducer(null, action)
  expect(actual).toBe(expected)
})

test('Skip should skip 1 round if no payload is supplied', () => {
  const expected = {
    ...waitingBidState,
    scores: genMap([0], [0], [0], [0]),
    currentPlayerOrder: ['b', 'c', 'd', 'a'],
    currentRound: 2
  }
  const state = {
    ...waitingBidState
  }
  const action = skipAction(1)
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Skip should reset bid and win data', () => {
  const expected = {
    ...waitingBidState,
    scores: genMap([0], [0], [0], [0]),
    currentPlayerOrder: ['b', 'c', 'd', 'a'],
    currentRound: 2
  }
  const state = {
    ...waitingWinState,
    bid: genMap(0, 1, 0, 1),
    win: genMap(0, 1, 0, 0)
  }
  const action = skipAction(1)
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Skip on last round should change state to ended', () => {
  const expected = {
    ...endedState,
    rounds: 1,
    scores: genMap([0], [0], [0], [0])
  }
  const state = {
    ...waitingBidState,
    rounds: 1
  }
  const action = skipAction(1)
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Skip on ended state should do no-op', () => {
  const expected = {
    ...endedState
  }
  const state = {
    ...endedState
  }
  const action = skipAction(1)
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Set bid should do no-op when state is null', () => {
  const expected = null
  const action = setBidAction(genMap(0, 0, 0, 0))
  const actual = reducer(null, action)
  expect(actual).toBe(expected)
})

test('Set bid should do no-op when stage is ended', () => {
  const expected = {
    ...endedState
  }
  const state = {
    ...endedState
  }
  const action = setBidAction(genMap(0, 0, 0, 0))
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Set bid should set bid property', () => {
  const expected = {
    ...waitingBidState,
    bid: genMap(0, 1, 0, 0)
  }
  const state = {
    ...waitingBidState,
    bid: genMap(0, 0, 0, 0)
  }
  const action = setBidAction(genMap(0, 1, 0, 0))
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Set win should do no-op when state is null', () => {
  const expected = null
  const action = setWinAction(genMap(0, 0, 0, 0))
  const actual = reducer(null, action)
  expect(actual).toBe(expected)
})

test('Set win should do no-op when stage is ended', () => {
  const expected = {
    ...endedState
  }
  const state = {
    ...endedState
  }
  const action = setWinAction(genMap(0, 0, 0, 0))
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Set win should set win property', () => {
  const expected = {
    ...waitingWinState,
    win: genMap(0, 1, 0, 0)
  }
  const state = {
    ...waitingWinState,
    win: genMap(0, 0, 0, 0)
  }
  const action = setWinAction(genMap(0, 1, 0, 0))
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Bid should do no-op when the state is null', () => {
  const expected = null
  const action = bidAction(genMap(0, 1, 0, 1))
  const actual = reducer(null, action)
  expect(actual).toBe(expected)
})

test('Bid should do no-op when game stage is ended', () => {
  const expected = {
    ...endedState
  }
  const state = {
    ...endedState
  }
  const action = bidAction(genMap(0, 1, 0, 1))
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Bid should change stage to waitingWin', () => {
  const expected = {
    ...waitingWinState,
    bid: genMap(0, 1, 0, 1)
  }
  const state = {
    ...waitingBidState,
    bid: genMap(0, 0, 0, 0)
  }
  const action = bidAction(genMap(0, 1, 0, 1))
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Bid should use data from currentGame.bid when bid is not given in payload', () => {
  const expected = {
    ...waitingWinState,
    bid: genMap(0, 1, 0, 1)
  }
  const state = {
    ...waitingBidState,
    bid: genMap(0, 1, 0, 1)
  }
  const action = bidAction()
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Win should do no-op when state is null', () => {
  const expected = null
  const state = null
  const action = winAction(genMap(0, 1, 0, 0))
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Win should do no-op when game stage is ended', () => {
  const expected = {
    ...endedState
  }
  const state = {
    ...endedState
  }
  const action = winAction(genMap(0, 1, 0, 0))
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Win should change stage to waitingBid', () => {
  const expected = {
    ...waitingBidState,
    currentPlayerOrder: ['b', 'c', 'd', 'a'],
    currentRound: 2,
    scores: genMap([11], [10], [10], [-1])
  }
  const state = {
    ...waitingWinState,
    bid: genMap(1, 0, 0, 1)
  }
  const action = winAction(genMap(1, 0, 0, 0))
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Win should change stage to ended for last round', () => {
  const expected = {
    ...endedState,
    rounds: 1,
    scores: genMap([11], [10], [10], [-1])
  }
  const state = {
    ...waitingWinState,
    rounds: 1,
    bid: genMap(1, 0, 0, 1)
  }
  const action = winAction(genMap(1, 0, 0, 0))
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Win should use data from gameStage.win if win is not given in action', () => {
  const expected = {
    ...waitingBidState,
    currentPlayerOrder: ['b', 'c', 'd', 'a'],
    currentRound: 2,
    scores: genMap([11], [10], [10], [-1])
  }
  const state = {
    ...waitingWinState,
    bid: genMap(1, 0, 0, 1),
    win: genMap(1, 0, 0, 0)
  }
  const action = winAction()
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Undo should do no-op when state is null', () => {
  const expected = null
  const state = null
  const action = undoAction()
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Undo should do no-op when stage is ended', () => {
  const expected = {
    ...endedState
  }
  const state = {
    ...endedState
  }
  const action = undoAction()
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Undo should do no-op when stage is waitingBid', () => {
  const expected = {
    ...waitingBidState
  }
  const state = {
    ...waitingBidState
  }
  const action = undoAction()
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Undo should roll back stage is waitingWin', () => {
  const expected = {
    ...waitingBidState,
    bid: genMap(0, 0, 0, 0)
  }
  const state = {
    ...waitingWinState,
    bid: genMap(0, 0, 0, 0)
  }
  const action = undoAction()
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Change players should do no-op in null state', () => {
  const expected = null
  const state = null
  const newNames = genMap('John', 'Mary', 'Henry', 'Joe')
  const action = changePlayersAction(newNames, 'a', 13)
  const actual = reducer(state, action)
  expect(actual).toBe(expected)
})

test('Change players should do no-op in ended state', () => {
  const expected = {
    ...endedState
  }
  const state = {
    ...endedState
  }
  const newNames = genMap('John', 'Mary', 'Henry', 'Joe')
  const action = changePlayersAction(newNames, 'a', 13)
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Change players should do no-op when given names are same as original map and maker is the current one', () => {
  const expected = {
    ...waitingBidState
  }
  const state = {
    ...waitingBidState
  }
  const newNames = genMap('John', 'Mary', 'Henry', 'Joe')
  const action = changePlayersAction(newNames, 'a', 13)
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Change players should revert stage to waitingBid for waitingWin stage', () => {
  const expected = {
    ...waitingBidState
  }
  const state = {
    ...waitingWinState
  }
  const newNames = genMap('John', 'Mary', 'Henry', 'Joe')
  const action = changePlayersAction(newNames, 'a', 13)
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test("Change players should change player's names from given map", () => {
  const expected = {
    ...waitingBidState,
    names: genMap('John', 'DPGJW', 'Henry', 'Joe')
  }
  const state = {
    ...waitingBidState
  }
  const newNames = genMap('John', 'DPGJW', 'Henry', 'Joe')
  const action = changePlayersAction(newNames, 'a', 13)
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Change players should change all maps order from given map (For no addition nor deletion)', () => {
  function genReorderedMap(b, c, d, a) {
    return {b, c, d, a}
  }

  const expected = {
    ...waitingBidState,
    names: genReorderedMap('Mary', 'Henry', 'Joe', 'John'),
    scores: genReorderedMap([], [], [], []),
    currentPlayerOrder: ['b', 'c', 'd', 'a'],
    bid: genReorderedMap(1, 0, 1, 0)
  }
  const state = {
    ...waitingBidState,
    bid: genMap(0, 1, 0, 1)
  }
  const newNames = genReorderedMap('Mary', 'Henry', 'Joe', 'John')
  const action = changePlayersAction(newNames, 'b', 13)
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Change players should re-order players by given maker', () => {
  const expected = {
    ...waitingBidState,
    currentPlayerOrder: ['c', 'd', 'a', 'b']
  }
  const state = {
    ...waitingBidState
  }
  const newNames = genMap('John', 'Mary', 'Henry', 'Joe')
  const action = changePlayersAction(newNames, 'c', 13)
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Change players should change rounds from given payload', () => {
  const expected = {
    ...waitingBidState,
    rounds: 12
  }
  const state = {
    ...waitingBidState
  }
  const newNames = genMap('John', 'Mary', 'Henry', 'Joe')
  const action = changePlayersAction(newNames, 'a', 12)
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Change players should remove all data from removed player', () => {
  function genSmallMap(a, b, c) {
    return {a, b, c}
  }

  const expected = {
    ...waitingBidState,
    names: genSmallMap('John', 'Mary', 'Henry'),
    scores: genSmallMap([], [], []),
    currentPlayerOrder: ['a', 'b', 'c'],
    bid: genSmallMap(0, 1, 0)
  }
  const state = {
    ...waitingBidState,
    bid: genMap(0, 1, 0, 1)
  }
  const newNames = genSmallMap('John', 'Mary', 'Henry')
  const action = changePlayersAction(newNames, 'a', 13)
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Change players should assign 0 mark for ended rounds of new player', () => {
  function genNewMap(a, b, c, d, e) {
    return {b, c, d, a, e}
  }

  const expected = {
    ...waitingBidState,
    currentRound: 2,
    names: genNewMap('John', 'Mary', 'Henry', 'Joe', 'DPGJW'),
    currentPlayerOrder: ['b', 'c', 'd', 'a', 'e'],
    scores: genNewMap([10], [11], [10], [-1], [0]),
    bid: genNewMap(0, 0, 0, 0, 0)
  }
  const state = {
    ...waitingBidState,
    currentRound: 2,
    currentPlayerOrder: ['b', 'c', 'd', 'a'],
    scores: genMap([10], [11], [10], [-1])
  }
  const newNames = genNewMap('John', 'Mary', 'Henry', 'Joe', 'DPGJW')
  const action = changePlayersAction(newNames, 'b', 13)
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})

test('Change players should end game when given rounds is less than current round', () => {
  const expected = {
    ...endedState,
    rounds: 1,
    scores: genMap([0], [0], [0], [0]),
    names: genMap('Mary', 'John', 'Henry', 'Joe')
  }
  const s = [0, 0] // Short hand for score
  const state = {
    ...waitingBidState,
    currentRound: 3,
    currentPlayerOrder: ['c', 'd', 'a', 'b'],
    scores: genMap(s, s, s, s)
  }
  const newNames = genMap('Mary', 'John', 'Henry', 'Joe')
  const action = changePlayersAction(newNames, 'a', 1)
  const actual = reducer(state, action)
  expect(actual).toEqual(expected)
})
