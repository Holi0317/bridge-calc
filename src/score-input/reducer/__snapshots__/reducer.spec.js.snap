// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Production of default basic state after start 1`] = `
Object {
  "bid": Object {
    "0": 0,
    "1": 0,
    "2": 0,
    "3": 0,
  },
  "currentPlayerOrder": Array [
    "0",
    "1",
    "2",
    "3",
  ],
  "currentRound": 1,
  "names": Object {
    "0": "John",
    "1": "Mary",
    "2": "Henry",
    "3": "Joe",
  },
  "rounds": 13,
  "scores": Object {
    "0": Array [],
    "1": Array [],
    "2": Array [],
    "3": Array [],
  },
  "stage": "waitingBid",
  "startTime": 1970-01-01T00:00:00.000Z,
}
`;

exports[`Start on second round should work 1`] = `
Object {
  "bid": Object {
    "4": 0,
    "5": 0,
    "6": 0,
    "7": 0,
  },
  "currentPlayerOrder": Array [
    "5",
    "6",
    "7",
    "4",
  ],
  "currentRound": 2,
  "names": Object {
    "4": "John",
    "5": "Mary",
    "6": "Henry",
    "7": "Joe",
  },
  "rounds": 13,
  "scores": Object {
    "4": Array [
      0,
    ],
    "5": Array [
      0,
    ],
    "6": Array [
      0,
    ],
    "7": Array [
      0,
    ],
  },
  "stage": "waitingBid",
  "startTime": 1970-01-01T00:00:00.000Z,
}
`;
